File: 526. Beautiful Arrangement/Java/Beautiful Arrangement.java
public class Solution {
    public int countArrangement(int N) {
        class Utils {
            int result = 0;

            void arrangement(int index, boolean[] used) {
                if (index == used.length) {
                    result++;
                    return;
                }
                for (int i = 1; i < used.length; i++) {
                    if (!used[i] && (i % index == 0 || index % i == 0)) {
                        used[i] = true;
                        arrangement(index + 1, used);
                        used[i] = false;
                    }
                }
            }
        }
        Utils utils = new Utils();
        utils.arrangement(1, new boolean[N + 1]);
        return utils.result;
    }
}

File: 526. Beautiful Arrangement/Python/Beautiful Arrangement.py
class Solution(object):
    def countArrangement(self, N):
        """
        :type N: int
        :rtype: int
        """
        self.result = 0

        def arrangement(index, used):
            if index == len(used):
                self.result += 1
                return
            for i in range(1, len(used)):
                if not used[i] and (i % index == 0 or index % i == 0):
                    used[i] = True
                    arrangement(index + 1, used)
                    used[i] = False

        arrangement(1, [False] * (N + 1))
        return self.result
